apps/ui/src/suite/api/ServerGliaswapAPI.ts

```diff
@@ -66,21 +111,65 @@ export class ServerGliaswapAPI implements GliaswapAPI {
     return Promise.resolve([]);
   }
 
-  async getAssetsWithBalance(lock: Script, assets?: ChainSpec[]): Promise<GliaswapAssetWithBalance[]> {
+  async getAssetsWithBalance(
+    lock: CkbScript,
+    assets?: Asset[],
+    ethAddr?: string,
+    web3?: Web3,
```

Comment:
we should move the dependencies to implement, for example `new GliaswapServerAndWeb3API(web3,...)`